Use native implementation of LZMA and XZ

hasenbanck opened the pull request:
This uses the native port of the XZ's liblzma. The main advantage is, that the crate is native Rust and is also used by the seven-zip2 crate.

Performance should be equal to the liblzma crate, since I spend quite a bit of time to improve both decoding and encoding performance.

I only used the single threaded version of the reader/writer, since parallelization was not enabled for liblzma. lzma-rust2 has multithreaded reader/writer though.

This PR also fixed the LZMA decoding. The old implementation actually didn't worked. The test file used STORE as compression. I checked the APPNOTE.txt to make sure that the properties are properly read. The test file is also updated (created with 7zip).

I had to change the signature of the public function `Decoder::new()`, since the LZMA decoder needs to know the uncompressed size to limit the range decoder. The LZMA stream CAN include an end-of-stream marker (EOS), but this is not guranteed. Most files out there have an EOS, but the specification allows LZMA streams without it (so it's best to limit the LZMA decoder with the uncompressed size). Another option 

I remembered that I had to defer the initialization of reading the properties when implementing PPMd, so I did this here too.

Edit: I will have a look into what breaks the `RefUnwindSafe` auto trait upstream.
Edit2:  `UnwindSafe` and `RefUnwindSafe` breaks because we store `std::io::Error` sometimes, which in itself is not  `UnwindSafe`. Have to think about thow to handle that.

Pr0methean commented on src/read.rs:
```diff
@@ -1901,12 +1911,18 @@ pub fn read_zipfile_from_stream<R: Read>(reader: &mut R) -> ZipResult<Option<Zip
     let limit_reader = reader.take(result.compressed_size);
 
     let result_crc32 = result.crc32;
+    let result_uncompressed_size = result.uncompressed_size;
     let result_compression_method = result.compression_method;
```
```suggestion
let { crc32, uncompressed_size, compression_method, .. } = result;
```

hasenbanck replied:
Good catch!

Pr0methean commented on Cargo.toml:
```diff
@@ -78,18 +78,16 @@ deflate-flate2-zlib = ["deflate-flate2", "flate2/zlib"]
 jiff-02 = ["dep:jiff"]
 nt-time = ["dep:nt-time"]
-lzma = ["dep:liblzma"]
-lzma-static = ["lzma", "liblzma/static"]
```
Keep `lzma-static` and `xz-static` for backward compatibility, even if they have no effects except enabling `lzma` and `xz`.

hasenbanck replied:
I will add them back.

hasenbanck commented:
@Pr0methean I fixed the issue with the broken UnwindSafe and RefUndindSafe and fixed the issues marked by the code review. I had to remove the test that the AI did, since it doesn't apply to lzma-rust2. The semver check action seems incorrect in the assasment of the missing default feature, since lzma provides the same functionality now.

As a note: #378 seems dangerous, since it doesn't guard against LZMA files that don't have an EOS and the memory limit handling seems very odd.

hasenbanck commented:
@Pr0methean Great! You can ping me if LZMA, XZ or PPMd related issues should pop up in the future.
